types/v56/types.go

```diff
@@ -466,6 +466,104 @@ type AdminVdc struct {
+	NsxTManagerReference            *Reference         `xml:"NsxTManagerReference,omitempty"`            // An optional reference to a registered NSX-T Manager to back networking operations for this provider VDC.
+	ResourcePoolRefs                *VimObjectRefs     `xml:"ResourcePoolRefs,omitempty"`                // Resource pools backing this provider VDC. On create, you must specify a resource pool that is not used by (and is not the child of a resource pool used by) any other provider VDC. On modify, this element is required for schema validation, but its contents cannot be changed.
+	VimServer                       *Reference         `xml:"VimServer,omitempty"`                       // The vCenter server that provides the resource pools and datastores. A valid reference is required on create. On modify, this element is required for schema validation, but its contents cannot be changed.
+}
+
+// VMWHostReferences represents a list of available hosts.
+// Type: VMWHostReferencesType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: Represents a list of available hosts.
+// Since: 1.0
+type VMWHostReferences struct {
+	HostReference []*Reference `xml:"HostReference,omitempty"`
+	Link          *Link        `xml:"Link,omitempty"`
+}
+
+// RootComputeCapacity represents compute capacity with units.
+// Type: RootComputeCapacityType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: Represents compute capacity with units.
+// Since: 0.9
+type RootComputeCapacity struct {
+	Cpu       *ProviderVdcCapacity `xml:"Cpu"`
+	IsElastic bool                 `xml:"IsElastic,omitempty"`
+	IsHA      bool                 `xml:"IsHA,omitempty"`
+	Memory    *ProviderVdcCapacity `xml:"Memory"`
+}
+
+// NetworkPoolReferences is a container for references to network pools in this vDC.
+// Type: NetworkPoolReferencesType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: Container for references to network pools in this vDC.
+// Since: 0.9
+type NetworkPoolReferences struct {
+	NetworkPoolReference []*Reference `xml:"NetworkPoolReference"`
+}
+
+// ProviderStorageProfiles is a container for references to storage profiles associated with a Provider vDC.
+// Type: ProviderVdcStorageProfilesType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: Container for references to storage profiles associated with a Provider vDC.
+// Since: 0.9
+type ProviderStorageProfiles struct {
+	ProviderVdcStorageProfile []*Reference `xml:"ProviderVdcStorageProfile"`
+}
+
+// ProviderVdcCapacity represents resource capacity in a Provider vDC.
+// Type: ProviderVdcCapacityType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// Description: Represents resource capacity in a Provider vDC.
+// Since: 0.9
+type ProviderVdcCapacity struct {
+	Allocation int64  `xml:"Allocation,omitempty"`
+	Overhead   int64  `xml:"Overhead,omitempty"`
+	Reserved   int64  `xml:"Reserved,omitempty"`
+	Total      int64  `xml:"Total"`
```

Comment:
Possibly this can also have `omitempty`?

Not critical as we will find this out when implementing missing CRUD operations in the future, but I'd suspect this could also be empty (is it a read-only field?).

Reply:
As per the docs, it is read-only and the only one required (apart from `Units`), so I was expecting to be always informed. But I can add it just in case.

Reply:
Added!